src/components/PullRequestList/index.tsx

```diff
@@ -26,10 +26,13 @@ const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
 
   const prUrl = `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/${prType}?page=${page}`;
 
+  const { CancelToken } = axios;
+  const source = CancelToken.source();
+
   const fetchPRs = async () => {
     try {
       setLoading(true);
-      const response = await axios.get(prUrl);
+      const response = await axios({ url: prUrl, cancelToken: source.token });
```

Comment:
We don't use `axios` or any request library in application code.

We have a [helper function](https://github.com/Real-Dev-Squad/website-status/blob/develop/src/helperFunctions/fetch.ts) for the same.

We also have a [custom hook](https://github.com/Real-Dev-Squad/website-status/blob/develop/src/hooks/useFetch.ts) to make requests.

You'll have to write this cancellation logic in both of them.

Reply:
Yes made all the necessary changes.